basyx/aas/adapter/http.py

```diff
@@ -585,6 +586,10 @@ def _get_submodel_reference(cls, aas: model.AssetAdministrationShell, submodel_i
     def _get_slice(cls, request: Request, iterator: Iterator[T]) -> Tuple[Iterator[T], int]:
         limit = request.args.get('limit', type=int, default=10)
         cursor = request.args.get('cursor', type=int, default=0)
+        limit_str= request.args.get('limit', type=str, default="10")
+        cursor_str= request.args.get('cursor', type=str, default="0")
+        if not limit_str.isdigit() or not cursor_str.isdigit():
+            raise BadRequest("Cursor and limit must be positive integers!")
```

Comment:
While the code is probably the shortest possible, I don't like that `limit` and `cursor` are retrieved twice, and also have the default values specified in two separate lines. What about the following instead:
```python
limit = request.args.get('limit', default="10")
cursor = request.args.get('cursor', default="0")
try:
    limit, cursor = int(limit), int(cursor)
    if limit < 0 or cursor < 0:
        raise ValueError
except ValueError:
    raise BadRequest("Cursor and limit must be positive integers!")
```

Reply:
I also like the above more 👍